applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_orthotropic_damage.cpp

```diff
@@ -0,0 +1,709 @@
+        const double c2 = c1 * (1.0 - nu); // Cii
+        const double c3 = c1 * nu;  // Cij
+        const double c4 = c1 * 0.5 * (1 - 2 * nu); // Gij
+        rSecantTensor(0,0) = (1.0 - rDamages[0]) * c2;
+        rSecantTensor(1,1) = (1.0 - rDamages[1]) * c2;
+        rSecantTensor(2,2) = (1.0 - rDamages[2]) * c2;
+        rSecantTensor(0,1) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c3;
+        rSecantTensor(0,2) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[2]))*c3;
+        rSecantTensor(1,0) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c3;
+        rSecantTensor(1,2) = std::sqrt((1.0 - rDamages[1])*(1.0 - rDamages[2]))*c3;
+        rSecantTensor(2,0) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[2]))*c3;
+        rSecantTensor(2,1) = std::sqrt((1.0 - rDamages[1])*(1.0 - rDamages[2]))*c3;
+        rSecantTensor(3,3) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c4;
+        rSecantTensor(4,4) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[2]))*c4;
+        rSecantTensor(5,5) = std::sqrt((1.0 - rDamages[1])*(1.0 - rDamages[2]))*c4;
+    } else { // 2D Case -> Plane strain
+        const double c0 = E / ((1.0 + nu)*(1.0 - 2.0 * nu));
+        const double c1 = (1.0 - nu)*c0;
+        const double c2 = c0 * nu;
+        const double c3 = (0.5 - nu)*c0;
+        rSecantTensor(0,0) = (1.0 - rDamages[0])*c1;
+        rSecantTensor(1,1) = (1.0 - rDamages[1])*c1;
+        rSecantTensor(0,1) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c2;
+        rSecantTensor(1,0) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c2;
+        rSecantTensor(2,2) = std::sqrt((1.0 - rDamages[0])*(1.0 - rDamages[1]))*c3;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+void GenericSmallStrainOrthotropicDamage<TConstLawIntegratorType>::CalculateRotationMatrix(
+    Matrix& rRotationTensor, 
+    Matrix rEigenVectorsMatrix,
+    const Matrix& rEigenValuesMatrix
+    )
+{
+    if (rRotationTensor.size1() != VoigtSize)
+        rRotationTensor.resize(VoigtSize,VoigtSize);
+    noalias(rRotationTensor) = ZeroMatrix(VoigtSize,VoigtSize);
+
+    if (Dimension == 3) {
+        // Reorder the principal stresses
+        double a = rEigenValuesMatrix(0,0);
+        double b = rEigenValuesMatrix(1,1);
+        double c = rEigenValuesMatrix(2,2);
+        int i=0,j=0,k=0;
+        if (a >= b && b >= c) {i=0; j=1; k=2;}
+        else if (a >= c && c >= b) {i=0; j=2; k=1;}
+        else if (b >= a && a >= c) {i=1; j=0; k=2;}
+        else if (b >= c && c >= a) {i=1; j=2; k=2;}
+        else if (c >= a && a >= b) {i=2; j=0; k=1;}
+        else if (c >= b && b >= a) {i=2; j=1; k=0;}
+        else KRATOS_ERROR << "Problem with the principal stresses..." << std::endl;
+
+        Matrix copy(VoigtSize,VoigtSize);
```

Comment:
```suggestion
        Matrix copy(rEigenVectorsMatrix);
```

Easier

Reply:
this can be done? OMG

Reply:
https://www.geeksforgeeks.org/copy-constructor-in-cpp/

Reply:
Is is called copy constructor

![imagen](https://user-images.githubusercontent.com/19991680/61235918-f1e68600-a736-11e9-9f5e-f9720cbe8957.png)

Reply:
![image](https://user-images.githubusercontent.com/26402605/61236014-3ffb8980-a737-11e9-924a-10db7aceb778.png)
I knew about the copy but not with the entity Matrix :)